opentelekomcloud/resource_opentelekomcloud_cce_node_pool_v3.go

```diff
@@ -0,0 +1,542 @@
+			},
+			"taints": {
+				Type:     schema.TypeList,
+				Optional: true,
+				ForceNew: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"key": {
+							Type:     schema.TypeString,
+							Required: true,
+						},
+						"value": {
+							Type:     schema.TypeString,
+							Required: true,
+						},
+						"effect": {
+							Type:     schema.TypeString,
+							Required: true,
+						},
+					}},
+			},
+			"key_pair": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				ForceNew:     true,
+				ExactlyOneOf: []string{"password", "key_pair"},
+			},
+			"password": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				ForceNew:     true,
+				Sensitive:    true,
+				ExactlyOneOf: []string{"password", "key_pair"},
+			},
+			"subnet_id": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+			},
+			"preinstall": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+				StateFunc: func(v interface{}) string {
+					switch v.(type) {
+					case string:
+						return installScriptHashSum(v.(string))
+					default:
+						return ""
+					}
+				},
+			},
+			"postinstall": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+				StateFunc: func(v interface{}) string {
+					switch v.(type) {
+					case string:
+						return installScriptHashSum(v.(string))
+					default:
+						return ""
+					}
+				},
+			},
+			"scale_enable": {
+				Type:     schema.TypeBool,
+				Optional: true,
+			},
+			"min_node_count": {
+				Type:     schema.TypeInt,
+				Optional: true,
+			},
+			"max_node_count": {
+				Type:     schema.TypeInt,
+				Optional: true,
+			},
+			"scale_down_cooldown_time": {
+				Type:     schema.TypeInt,
+				Optional: true,
+			},
+			"priority": {
+				Type:     schema.TypeInt,
+				Optional: true,
+			},
+			"server_group_reference": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+			},
+			"status": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func resourceCCENodePoolUserTags(d *schema.ResourceData) []tags.ResourceTag {
+	tagRaw := d.Get("user_tags").(map[string]interface{})
+	return expandResourceTags(tagRaw)
+}
+
+func resourceCCENodePoolV3Create(d *schema.ResourceData, meta interface{}) error {
+	config := meta.(*Config)
+	nodePoolClient, err := config.cceV3Client(GetRegion(d, config))
+	if err != nil {
+		return fmt.Errorf("error creating OpenTelekomCloud CCE Node Pool client: %s", err)
+	}
+
+	var base64PreInstall, base64PostInstall string
+	if v, ok := d.GetOk("preinstall"); ok {
+		base64PreInstall = installScriptEncode(v.(string))
+	}
+	if v, ok := d.GetOk("postinstall"); ok {
+		base64PostInstall = installScriptEncode(v.(string))
+	}
+	var loginSpec nodes.LoginSpec
+	if hasFilledOpt(d, "key_pair") {
+		loginSpec = nodes.LoginSpec{SshKey: d.Get("key_pair").(string)}
+	} else if hasFilledOpt(d, "password") {
```

Comment:
Why `else` is needed?

Reply:
because user can use password?

Reply:
I mean, you can use `if` here without `else`